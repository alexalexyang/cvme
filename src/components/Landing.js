import React, { useState } from "react";
import { getGitHubProfile } from "./utils/githubProfileGraphQL";
import CV from "./CV";

function Landing() {
  const [userLogin, setUserLogin] = useState("");
  const [user, setUser] = useState();

  const submitUserLogin = async e => {
    e.preventDefault();
    userLogin && getGitHubProfile(userLogin, setUser);
  };

  const gitHubUsernameForm = () => {
    return (
      <div className="section page-wrapper">
        <div className="container form-wrapper">
          <form className="form field has-addons" onSubmit={submitUserLogin}>
            <div className="control has-icons-left">
              <input
                className="input"
                type="text"
                name="github-username"
                placeholder="What's your GitHub username?"
                onChange={e => setUserLogin(e.target.value)}
              />
              <span className="icon is-small is-left">
                <svg
                  className="rotate"
                  width="24"
                  height="24"
                  fill="silver"
                  xmlns="http://www.w3.org/2000/svg"
                  fillRule="evenodd"
                  clipRule="evenodd"
                >
                  <path d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm7.753 18.305c-.261-.586-.789-.991-1.871-1.241-2.293-.529-4.428-.993-3.393-2.945 3.145-5.942.833-9.119-2.489-9.119-3.388 0-5.644 3.299-2.489 9.119 1.066 1.964-1.148 2.427-3.393 2.945-1.084.25-1.608.658-1.867 1.246-1.405-1.723-2.251-3.919-2.251-6.31 0-5.514 4.486-10 10-10s10 4.486 10 10c0 2.389-.845 4.583-2.247 6.305z" />{" "}
                </svg>
              </span>
            </div>
            <div className="control">
              <button className="button">Submit</button>
            </div>
          </form>
        </div>
      </div>
    );
  };

  return user ? <CV user={user} /> : gitHubUsernameForm();
}

export default Landing;
