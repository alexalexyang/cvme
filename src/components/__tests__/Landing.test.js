import Landing from "../Landing";

describe("<Landing />", () => {
  it("Snapshot test for <Landing />", () => {
    const component = renderer.create(<Landing />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
