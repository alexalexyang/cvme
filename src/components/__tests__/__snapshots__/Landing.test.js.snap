// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> Snapshot test for <Landing /> 1`] = `
<div
  className="section page-wrapper"
>
  <div
    className="container form-wrapper"
  >
    <form
      className="form field has-addons"
      onSubmit={[Function]}
    >
      <div
        className="control has-icons-left"
      >
        <input
          className="input"
          name="github-username"
          onChange={[Function]}
          placeholder="What's your GitHub username?"
          type="text"
        />
        <span
          className="icon is-small is-left"
        >
          <svg
            className="rotate"
            clipRule="evenodd"
            fill="silver"
            fillRule="evenodd"
            height="24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm7.753 18.305c-.261-.586-.789-.991-1.871-1.241-2.293-.529-4.428-.993-3.393-2.945 3.145-5.942.833-9.119-2.489-9.119-3.388 0-5.644 3.299-2.489 9.119 1.066 1.964-1.148 2.427-3.393 2.945-1.084.25-1.608.658-1.867 1.246-1.405-1.723-2.251-3.919-2.251-6.31 0-5.514 4.486-10 10-10s10 4.486 10 10c0 2.389-.845 4.583-2.247 6.305z"
            />
             
          </svg>
        </span>
      </div>
      <div
        className="control"
      >
        <button
          className="button"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
